### ISSUE TITLE
List of minor issues and fixes for pybind 3.0

### ISSUE THREAD

**Issue by YannickJadoul** [pybind/pybind11#2609]
pybind 3.0 might still take a while, but it's good to start thinking what API-breaking changes we need to introduce, by that time. I've created a milestone v3.0.0, but to reduce cluttering the issues, minor things can also be commented and listed here:

- [ ] Remove deprecations
  - [ ] `PYBIND11_PLUGIN`
  - [ ] `object::operator==`

Suggestions and additions very welcome!

**Comment by bstaletic** [comment]
Should we change `object::operator==` to mean `object::equals`?

**Comment by YannickJadoul** [comment]
> Should we change `object::operator==` to mean `object::equals`?

Or that. But that might be dangerous? I mainly just wanted to list sómething; all of this can still be dicussed ;-)

**Comment by sizmailov** [comment]
I wounder if #2244 has any chance to get into v3.0.

**Comment by YannickJadoul** [comment]
> I wounder if #2244 has any chance to get into v3.0.

When it comes to changes to the API, sure. However, it seems like there were also objections by @wjakob against the complexity added by this?

**Comment by sizmailov** [comment]
Sorry for bringing it up once again after definitive "NO" from Jakob. I was hoping it has a bit more chances in context of v3.0.

(I believe the complexity associated with the PR is the minimum that required to deliver the feature. Namely, it's only type caster specializations + verbose explicit types (for C++-11 compatibility). If it's still too much, let be it. In this case it might (should) be read as "annotation-related features are not welcome until v4.0.")

**Comment by YannickJadoul** [comment]
> Sorry for bringing it up once again after definitive "NO" from Jakob. I was hoping it has a bit more chances in context of v3.0.
> 
> (I believe the complexity associated with the PR is the minimum that required to deliver the feature. Namely, it's only type caster specializations + verbose explicit types (for C++-11 compatibility). If it's still too much, let be it. In this case it might (should) be read as "annotation-related features are not welcome until v4.0.")

No, don't worry. I would maybe find it interesting, but I'm just warning you that this one might be more tricky than just "Yes, bring it on, now that we can break stuff!" :-) It doesn't look all too complex to me, but then that's just me, ofc :-)

**Comment by shelllet** [comment]
it's good idea to support std::any

**Comment by shelllet** [comment]
Sub-interpreter support
